### PR TITLE
Added 1) floating point expontent (e^x) ...

### PR DESCRIPTION
**Pull request by jbaileyhandle**
... 2) FP to int conversion and 3)iterative fixed-point divsion as a synthesis-friendly alternative to the / operator.

### CODE REVIEW

**Comment by jbaileyhandle** [comment]
Moved exponent to third_party/xls_go_math, incorporated division in third_party/xls_berkeley_softfloat.

File: third_party/xls_go_math/fpexp_32.x [hunk 1]
```diff
@@ -0,0 +1,244 @@
+// Copyright 2021 The XLS Authors
```
Comment (meheff) [inline third_party/xls_go_math/fpexp_32.x:1]: This should be licensed like this file:

https://github.com/google/xls/blob/main/third_party/xls_go_math/fpsqrt_32.x

No "Licensed under the Apache License...". Same with other files.

File: third_party/xls_go_math/fpexp_32_test.cc [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+ABSL_FLAG(bool, use_opt_ir, true, "Use optimized IR.");
+ABSL_FLAG(int, num_threads, 0,
+          "Number of threads to use. Set to 0 to use all.");
+ABSL_FLAG(int64_t, num_samples, 1024 * 1024,
+          "Number of random samples to test.");
+
+namespace xls {
+
+constexpr const char kOptIrPath[] = "xls/modules/fpexp_32.opt.ir";
```
Comment (meheff) [inline third_party/xls_go_math/fpexp_32_test.cc:39]: unused constants

File: xls/modules/fpldexp_32_test.cc [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+ABSL_FLAG(bool, use_opt_ir, true, "Use optimized IR.");
+ABSL_FLAG(int, num_threads, 0,
+          "Number of threads to use. Set to 0 to use all.");
+ABSL_FLAG(int64_t, num_samples, 1024 * 1024,
+          "Number of random samples to test.");
+
+namespace xls {
+
+constexpr const char kOptIrPath[] = "xls/modules/fpldexp_32.opt.ir";
```
Comment (meheff) [inline xls/modules/fpldexp_32_test.cc:39]: unused constants.

File: third_party/xls_go_math/fpexp_32.x [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+
+  fpldexp_32::fpldexp_32(y, k)
+}
+
+// Returns e^x
+pub fn fpexp_32(x: F32) -> F32 {
+  const ZERO = float32::zero(u1:0);
+
+  // Flush subnormals.
+  let x = float32::zero(u1:0)
```
Comment (meheff) [inline third_party/xls_go_math/fpexp_32.x:175]: we should probably have a flush_to_zero function somewhere. No need for you to do it though.

File: third_party/xls_go_math/fpexp_32.x [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+              if (float32::is_inf(x) && x.sign == u1:1)
+              else result;
+  result
+}
+
+
+#![test]
+fn fpexp_32_test() {
+  // Special cases.
+  // let _ = assert_eq(float32::qnan(), fpexp_32(float32::qnan()));
```
Comment (meheff) [inline third_party/xls_go_math/fpexp_32.x:229]: uncomment these tests?

**Review by meheff — CHANGES_REQUESTED** [review]
Thanks again for adding all this!

File: third_party/xls_go_math/fpexp_32.x (hunk 1 above)
Comment (jbaileyhandle) [reply third_party/xls_go_math/fpexp_32.x:1]: Done.
fpexp_32.x is the only file in this PR I can see where this change needs to be made - am I missing others?

File: third_party/xls_go_math/fpexp_32_test.cc (hunk 2 above)
Comment (jbaileyhandle) [reply third_party/xls_go_math/fpexp_32_test.cc:39]: Done.

File: xls/modules/fpldexp_32_test.cc (hunk 3 above)
Comment (jbaileyhandle) [reply xls/modules/fpldexp_32_test.cc:39]: Done.

File: third_party/xls_go_math/fpexp_32.x (hunk 4 above)
Comment (jbaileyhandle) [reply third_party/xls_go_math/fpexp_32.x:175]: Done (looks like it was already the float library).

File: third_party/xls_go_math/fpexp_32.x (hunk 5 above)
Comment (jbaileyhandle) [reply third_party/xls_go_math/fpexp_32.x:229]: Done.

**Review by meheff — APPROVED** [review]

**Pull request merged**